_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,983 @@
+
+<details closed markdown="block">
+  <summary>
+    Test it
+  </summary>
+  {: .text-delta}
+
+Search for the deployed model by providing its ID in the request:
+
+```json
+GET /_plugins/_ml/models/aVeif4oB5Vm0Tdw8zYO2
+```
+{% include copy-curl.html %}
+
+The response shows the model state as `DEPLOYED`:
+
+```json
+{
+  "name": "huggingface/sentence-transformers/msmarco-distilbert-base-tas-b",
+  "model_group_id": "Z1eQf4oB5Vm0Tdw8EIP2",
+  "algorithm": "TEXT_EMBEDDING",
+  "model_version": "1",
+  "model_format": "TORCH_SCRIPT",
+  "model_state": "DEPLOYED",
+  "model_content_size_in_bytes": 266352827,
+  "model_content_hash_value": "acdc81b652b83121f914c5912ae27c0fca8fabf270e6f191ace6979a19830413",
+  "model_config": {
+    "model_type": "distilbert",
+    "embedding_dimension": 768,
+    "framework_type": "SENTENCE_TRANSFORMERS",
+    "all_config": """{"_name_or_path":"old_models/msmarco-distilbert-base-tas-b/0_Transformer","activation":"gelu","architectures":["DistilBertModel"],"attention_dropout":0.1,"dim":768,"dropout":0.1,"hidden_dim":3072,"initializer_range":0.02,"max_position_embeddings":512,"model_type":"distilbert","n_heads":12,"n_layers":6,"pad_token_id":0,"qa_dropout":0.1,"seq_classif_dropout":0.2,"sinusoidal_pos_embds":false,"tie_weights_":true,"transformers_version":"4.7.0","vocab_size":30522}"""
+  },
+  "created_time": 1694482261832,
+  "last_updated_time": 1694482324282,
+  "last_registered_time": 1694482270216,
+  "last_deployed_time": 1694482324282,
+  "total_chunks": 27,
+  "planning_worker_node_count": 1,
+  "current_worker_node_count": 1,
+  "planning_worker_nodes": [
+    "4p6FVOmJRtu3wehDD74hzQ"
+  ],
+  "deploy_to_all_nodes": true
+}
+```
+
+You can also receive statistics for all deployed models in your cluster by sending a Models Profile API request:
+
+```json
+GET /_plugins/_ml/profile/models
+```
+</details>
+
+## Step 2: Ingest data with neural search
+
+Neural search uses a language model to transform text into vector embeddings. During ingestion, neural search creates vector embeddings for the text fields in the request. During search, you can generate vector embeddings for the query text by applying the same model, allowing you to perform vector similarity search on the documents.
+
+### Step 2(a): Create an ingest pipeline for neural search
+
+The first step in setting up [neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/) is to create an [ingest pipeline]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/). The ingest pipeline will contain one processor: a task that transforms document fields. For neural search, you'll need to set up a `text_embedding` processor that takes in text and creates vector embeddings from that text. You'll need the `model_id` of the model you set up in the previous section and a `field_map`, which specifies the name of the field from which to take the text (`text`) and the name of the field in which to record embeddings (`passage_embedding`):
```

Comment:
Maybe we should rephrase this? I feel like this isn't the first step as deploying a model was our first few steps. Maybe we can say, "Now that we have a model we can use that to configure neural search."